src/ray/core_worker/context.h

```diff
@@ -150,8 +151,12 @@ class WorkerContext {
   // Whether or not we should implicitly capture parent's placement group.
   bool placement_group_capture_child_tasks_ ABSL_GUARDED_BY(mutex_);
   // The runtime env for the current actor or task.
+  // For one worker context, it should have exactly one serialized runtime env; cache the
+  // parsed json and string for reuse.
+  std::string serialized_runtime_env_ ABSL_GUARDED_BY(mutex_);
   std::shared_ptr<nlohmann::json> runtime_env_ ABSL_GUARDED_BY(mutex_);
   // The runtime env info.
+  // For one worker context, it should be assigned only once.
```

Comment:
```suggestion
  // For one worker context, it should be assigned only once because Ray currently doesn't reuse worker to run tasks or actors with different runtime envs.
```

Reply:
updated